src/Main.purs

```diff
@@ -231,13 +232,9 @@ copyFile from to = do
   makeAff \k -> do
     src <- createReadStream from
     dst <- createWriteStream to
-    res <- Stream.pipe src dst
-    Stream.onError src (k <<< Left)
-    Stream.onError dst (k <<< Left)
-    Stream.onError res (k <<< Left)
-    Stream.onFinish res (k (Right unit))
+    Stream.pipe src dst
+    src # on_ Stream.errorH (k <<< Left)
+    dst # on_ Stream.errorH (k <<< Left)
```

Comment:
I think I would move the error stuff above the `pipe` line.

Otherwise, Node docs state:
> One important caveat is that if the Readable stream emits an error during processing, the Writable destination is not closed automatically. If an error occurs, it will be necessary to manually close each stream in order to prevent memory leaks.

So, the `src` errorH needs to call close on `dst` before calling `k <<< Left`.

Reply:
How would that look like?

Reply:
```purs
src # on_ Stream.errorH \err -> do
  Stream.destroy' dst $ error $ "Got error in src stream: " <> message err
  k $ Left err
dst # on_ Stream.errorH (k <<< Left)
Stream.pipe src dst

Reply:
Thanks!